### DISCUSSION TITLE
Firehose API

### DISCUSSION THREAD

**Discussion by CharlesDardaman** [bluesky-social/atproto#804] [How to do it?] [answered]
Is there currently a way to get a stream of all posts? I took a look in the golang code which seems to have some level of streaming all data [here](https://github.com/bluesky-social/indigo/tree/main/cmd/bigsky), but without documentation I'm going to need to dig a lot further to get it working.

**Comment by dholms** [comment, marked as answer]
There is. It's not well documented yet, but if you're down to get your hands a little dirty, you can start a websockets connection at: `/xrpc/com.atproto.sync.subscribeRepos`

**Reply by bnewbold** [reply]
We've got better docs today at: https://atproto.com/specs/event-stream
